### PR TITLE
hatsu: init at 0.2.0

### LINKED ISSUE #320169: Package request: Hatsu
**Project description**

Self-hosted & Fully-automated ActivityPub Bridge for Static Sites.

**Metadata**

* homepage URL: https://github.com/importantimport/hatsu
* source URL: https://github.com/importantimport/hatsu
* license: agpl3Only
* platforms: linux

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

### PR DESCRIPTION
**Pull request by kwaa**
## Description of changes

add `hatsu` package.

closed #320169

**Project description**

Self-hosted & Fully-automated ActivityPub Bridge for Static Sites.

**Metadata**

* homepage URL: https://github.com/importantimport/hatsu
* source URL: https://github.com/importantimport/hatsu
* license: agpl3Only
* platforms: linux

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/ha/hatsu/package.nix [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+, pkg-config
+, rustPlatform
+}:
+rustPlatform.buildRustPackage rec {
+  pname = "hatsu";
+  version = "0.2.0";
+
+  src = fetchFromGitHub {
+    owner = "importantimport";
+    repo = pname;
```
Comment (eclairevoyant) [inline pkgs/by-name/ha/hatsu/package.nix:15]:
```suggestion
    repo = "hatsu";
```

File: pkgs/by-name/ha/hatsu/package.nix [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+
+  env = { OPENSSL_NO_VENDOR = true; };
+
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+    ignoredVersions = "beta";
+  };
+
+  meta = with lib; {
+    description = "Self-hosted & Fully-automated ActivityPub Bridge for Static Sites.";
```
Comment (eclairevoyant) [inline pkgs/by-name/ha/hatsu/package.nix:33]: description shouldn't end in a period (per guidelines)
also title case is a bit strange here

```suggestion
    description = "Self-hosted and fully-automated ActivityPub bridge for static sites";
```

File: pkgs/by-name/ha/hatsu/package.nix [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  buildInputs = [ openssl ];
+
+  env = { OPENSSL_NO_VENDOR = true; };
+
+  passthru.updateScript = gitUpdater {
+    rev-prefix = "v";
+    ignoredVersions = "beta";
+  };
+
+  meta = with lib; {
```
Comment (eclairevoyant) [inline pkgs/by-name/ha/hatsu/package.nix:32]:
```suggestion
  meta = {
```

tracking issue: #292468

File: pkgs/by-name/ha/hatsu/package.nix [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+    description = "Self-hosted & Fully-automated ActivityPub Bridge for Static Sites.";
+    homepage = "https://github.com/importantimport/hatsu";
+    license = licenses.agpl3Only;
+    mainProgram = pname;
+    maintainers = with maintainers; [ kwaa ];
+    platforms = platforms.linux;
```
Comment (eclairevoyant) [inline pkgs/by-name/ha/hatsu/package.nix:38]:
```suggestion
    license = lib.licenses.agpl3Only;
    mainProgram = "hatsu";
    maintainers = with lib.maintainers; [ kwaa ];
    platforms = lib.platforms.linux;
```

**Review by eclairevoyant — DISMISSED** [review]
Thanks for the contribution; some feedback below:

**Comment by kwaa** [comment]
> Thanks for the contribution; some feedback below:

Thanks for the review, I've made the suggested changes.

**Review by eclairevoyant — COMMENTED** [review]
builds fine, but fails at runtime; not sure how this is meant to be tested/used
```
2024-06-16T04:41:12.917607Z  INFO hatsu: Hatsu v0.2.0 "celluloid"
2024-06-16T04:41:12.917618Z  INFO hatsu: loading environment variables
Error: AppError { error: "environment variable not found", error_id: 2373cf99-cfee-4c07-af68-bee029a69843, error_details: None, status: 500, context: SpanTrace [] }
```

**Comment by kwaa** [comment]
> builds fine, but fails at runtime; not sure how this is meant to be tested/used

The variables `HATSU_PRIMARY_ACCOUNT` and `HATSU_DOMAIN` need to be set. This is a simple example:

```bash
HATSU_PRIMARY_ACCOUNT="kwaa.dev" HATSU_DOMAIN="hatsu.local" nix run .#hatsu
```

**Comment by onny** [comment]
Result of `nixpkgs-review pr 320173` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>hatsu</li>
  </ul>
</details>

**Comment by onny** [comment]
Thank you!

**Pull request merged**